server/__init__.py

```diff
@@ -125,21 +118,14 @@ def validation_func(lobby_conn):
                 def validation_func(lobby_conn):
                     return lobby_conn.player.id not in game.host.foes
 
-            tasks.append(ctx.broadcast(
+            ctx.write_broadcast(
                 message,
                 lambda lobby_conn: lobby_conn.authenticated and validation_func(lobby_conn)
-            ))
-
-        try:
-            await asyncio.gather(*tasks)
-        except Exception as e:
-            logging.getLogger().exception(e)
-
-    ping_msg = encode_message('PING')
```

Comment:
What about exception processing? How they will behave?

Reply:
Hm. `send` doesn't raise errors as much as `drain` but I guess there are still some situations where it could raise an error.

https://github.com/python/cpython/blob/2477aed12d409c78a16ec435e01cee235af53221/Lib/asyncio/selector_events.py#L888

Reply:
Actually, I've already wrapped all calls to `StreamWriter.write` in an exception handler so I think the main issue would be if there is a message encoding error.

Reply:
Let's see :-)